core/src/main/java/com/scalar/db/transaction/consensuscommit/ParallelExecutor.java

```diff
@@ -47,81 +51,168 @@ public ParallelExecutor(ConsensusCommitConfig config) {
     this.parallelExecutorService = parallelExecutorService;
   }
 
-  public void prepare(List<ParallelExecutorTask> tasks) throws ExecutionException {
+  public void prepare(List<ParallelExecutorTask> tasks, String transactionId)
+      throws ExecutionException {
     try {
-      executeTasks(tasks, config.isParallelPreparationEnabled(), false);
+      // when the parallel preparation disabled, we stop running the tasks when one of them fails.
+      // when not, we need to wait for all the tasks to finish even if some of them fail
+      boolean stopOnError = !config.isParallelPreparationEnabled();
```

Comment:
In the preparation phase, we should switch `stopOnError` depending on whether the parallel preparation disabled or not.